lms/static/scripts/frontend_apps/components/URLPicker.tsx

```diff
@@ -74,9 +99,10 @@ export default function URLPicker({
         </Button>,
         <Button
           data-testid="submit-button"
+          form={formId}
           key="submit"
-          onClick={submit}
           variant="primary"
+          type="submit"
```

Comment:
This makes form submission always go through the form's "submit" event, so submission works consistently.